influence_benchmark/retroactive_evaluator/run_retroactive_evals.py

```diff
@@ -0,0 +1,177 @@
+import multiprocessing as mp
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+
+from influence_benchmark.backend.openai_backend import OpenAIBackend
+from influence_benchmark.retroactive_evaluator.hf_retroactive_evaluator import HFRetroactiveEvaluator
+from influence_benchmark.retroactive_evaluator.openai_retroactive_evaluator import OpenAIRetroactiveEvaluator
+from influence_benchmark.retroactive_evaluator.plot_retroactive_evals import metrics_by_run
+from influence_benchmark.utils.utils import find_freest_gpus, save_pickle
+
+
+def evaluate_single_run_gpt(
+    run: str,
+    backend_config: Dict[str, Any],
+    env_config_path: Path,
+    max_trajs_per_env: int,
+    max_iter: Optional[int],
+    load: bool,
+    save: bool,
+    run_dir_prefix: Path,
+):
+    # Note that the reason we load separate backends within the processes is because we
+    # otherwise get a pickling error when trying to parallelize across runs.
+    print(f"Evaluating run {run}.")
+    run_dir = run_dir_prefix / run
+    metrics = metrics_by_run(run)
+
+    # Initialize the backend within the process
+    backend = OpenAIBackend(**backend_config)
+
+    evaluator = OpenAIRetroactiveEvaluator(
+        run_path=run_dir,
+        backend_config=backend_config,
+        metrics=metrics,
+        env_config_path=env_config_path,
+        max_trajs_per_env=max_trajs_per_env,
+        backend=backend,
+    )
+
+    results_df = evaluator.evaluate_run(load=load, save=save, max_iter=max_iter)
+
+    save_name = run + "_gpt"
+    print(f"Saving results_df as {save_name}.pkl")
+    save_pickle(results_df, f"{save_name}.pkl")
+
+
+def evaluate_runs_gpt(
+    runs: List[str],
+    backend_config: Dict[str, Any],
+    env_config_path: Path,
+    max_trajs_per_env: int,
+    max_iter: Optional[int],
+    load: bool,
+    save: bool,
+    run_dir_prefix: Path,
+) -> Dict[str, pd.DataFrame]:
+    results_df_dict = {}
+    processes = []
+    mp.set_start_method("spawn")
+
+    for run in runs:
+        print(f"Starting process for run {run}.")
+        p = mp.Process(
```

Comment:
I think it's a mistake to do gpt stuff with mp, this task is not cpu bound. It should be simpler to just pass all messages from all runs to the same gpt backend straight away with async